sdk/src/wallet/operations/transaction/options.rs

```diff
@@ -1,43 +1,65 @@
-// Copyright 2021 IOTA Stiftung
+// Copyright 2024 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
+use alloc::collections::{BTreeMap, BTreeSet};
+
 use serde::{Deserialize, Serialize};
 
 use crate::{
     client::api::input_selection::Burn,
     types::block::{
         address::Address,
         context_input::ContextInput,
-        mana::ManaAllotment,
-        output::OutputId,
+        output::{AccountId, OutputId},
         payload::{signed_transaction::TransactionCapabilities, tagged_data::TaggedDataPayload},
     },
 };
 
 /// Options for transactions
-#[derive(Debug, Clone, Serialize, Deserialize, Default)]
+#[derive(Debug, Clone, Serialize, Deserialize)]
 #[serde(rename_all = "camelCase")]
+#[serde(default)]
 pub struct TransactionOptions {
-    #[serde(default)]
+    /// The strategy applied for base coin remainders.
     pub remainder_value_strategy: RemainderValueStrategy,
-    #[serde(default)]
+    /// An optional tagged data payload.
     pub tagged_data_payload: Option<TaggedDataPayload>,
-    #[serde(default)]
-    pub context_inputs: Option<Vec<ContextInput>>,
-    // If custom inputs are provided only they are used. If also other additional inputs should be used,
-    // `mandatory_inputs` should be used instead.
-    #[serde(default)]
-    pub custom_inputs: Option<Vec<OutputId>>,
-    #[serde(default)]
-    pub mandatory_inputs: Option<Vec<OutputId>>,
+    /// Transaction context inputs to include.
+    pub context_inputs: Vec<ContextInput>,
+    /// Inputs that must be used for the transaction.
+    pub mandatory_inputs: BTreeSet<OutputId>,
```

Comment:
So what do we think about adding `forbidden_inputs` as well?
custom_inputs originally also made it possible to prevent ISA from selecting some inputs that might should stay unspent for whatever reason
If we add `forbidden_inputs` here, this would basically solve https://github.com/iotaledger/iota-sdk/issues/1549 (the proposed way there might be nicer to use, but having this field can also do it)

Reply:
We could! Technically it's still possible the same way as before, by adding mandatory inputs and then disabling additional input selection, but your suggestion would be more flexible for sure. Alternatively, we could provide a filter closure argument and just call `retain` on the available inputs.

Reply:
> and then disabling additional input selection

How can it be disabled?

Reply:
Closure sounds annoying with bindings

Reply:
you can siable it with the new `allow_additional_input_selection` flag. And yes, that would not work well with bindings, you are right haha

Reply:
Ah thanks, missed the addition of  `allow_additional_input_selection`

Reply:
No need to add forbidden_inputs then in this PR, could still be done as part of the other issue